regions/shapes/tests/test_common.py

```diff
@@ -45,41 +45,53 @@ def test_contains_compared_to_shapely(self):
         np.random.seed(12345)
         x = np.random.uniform(self.sample_box[0], self.sample_box[1], 1000)
         y = np.random.uniform(self.sample_box[2], self.sample_box[3], 1000)
-        inside = self.reg.contains(PixCoord(x, y))
-        reg_shapely = self.reg.to_shapely()
-        inside_shapely = [reg_shapely.contains(Point(x[i], y[i])) for i in range(len(x))]
-        assert_equal(inside, inside_shapely)
+        try:
+            inside = self.reg.contains(PixCoord(x, y))
+            reg_shapely = self.reg.to_shapely()
+            inside_shapely = [reg_shapely.contains(Point(x[i], y[i])) for i in range(len(x))]
+            assert_equal(inside, inside_shapely)
+        except NotImplementedError:
+            pytest.skip()
```

Comment:
For these skip tests, you might do
```
if not HAS_SHAPELY:
     pytest.skip()
else:
    raise Exception("Got a 'not implemented error' but expected it to be implemented.")
```
or something along those lines.  I'm not sure this is necessary; let me know what you think.

Reply:
Ahh, I see, so this is meant to catch a subset of the possible tests that will be run?  You should include a note (a comment) about that, because it's not obvious

Reply:
yes, the `HAS SHAPELY` checks whether the system has the shapely module.
the try catch checks whether `to_shapely` is implemented for a particular class.
will add a comment for that.